File: solutions/clojure/darts/1/src/darts.clj
(ns darts)

(defn score
  "Calculates the score of a dart throw"
  [x y]
  ;; function body
  (let [d (+ (* x x) (* y y))]        
      (cond
        (<= d 1) 10
        (<= d 25) 5
        (<= d 100) 1
        :else 0)
    )
  )

File: solutions/dart/darts/1/lib/darts.dart
class Darts {
  int score(double x, double y){
    var ds = x*x + y*y;
    int score = 0;
    switch(ds){
      case >=0 && <=1 : score=10;
      case >1 && <=25 : score=5;
      case >25 && <=100 : score=1;
      default: score = 0;
    }
    return score;
  }
}
